### ISSUE TITLE
Do something with hook results

### ISSUE THREAD

**Issue by wmertens** [webpack/tapable#47]
In my app, I would like to return early if one of the hooks returns some specific value. Would that be possible?

So for example `hook.callUntil({args: [...], condition: result => result.foo})`.

**Comment by ooflorent** [comment]
You could use a `do/while` loop for that.

**Comment by wmertens** [comment]
@oofloren how would that work? There is no way to iterate hook subscribers, is there?

**Comment by ooflorent** [comment]
Sorry, the notification got lost!

```js
let result
do {
  result = hook.call(…)
} while (result.foo)
```

We use a similar technique in webpack:

https://github.com/webpack/webpack/blob/3024078e1e8ec3df00020acad2e2de8b199b27fc/lib/Compilation.js#L828-L834

**Comment by wmertens** [comment]
No, that calls *all* subscribers. The idea is to stop after the first subscriber that returns something specific.

is that what `callTapsSeries` does, can that be called directly?

**Comment by ooflorent** [comment]
I think all `call*` methods are private APIs. They are designed to be called by the code factory. Not user-land code.

**Comment by sokra** [comment]
> The idea is to stop after the first subscriber that returns something specific.

The `Bail` hooks stop after the first plugin that returns something.

**Comment by wmertens** [comment]
@sokra 	but that still doesn't allow checking for a specific value, and there's no way to wrap the hooks…

**Comment by MoonYaph** [comment]
@wmertens you can use bail hook if the the value is just what you need, just return it.

**Comment by wmertens** [comment]
@MoonYaph that would require wrapping all the subscribers to a hook and there's no way to do that.

**Comment by Telokis** [comment]
@wmertens Hi!

Did you manage to find a solution to this issue?
I'd like to do the same thing, as well.

**Comment by wmertens** [comment]
@Telokis I ended up rewriting it with a similar API but without all the browser optimizations.

https://github.com/StratoKit/strato-runner/blob/new-new-config/packages/launcher/src/tapable.js

**Comment by alexander-akait** [comment]
Let's close very old
